### PR TITLE
Fix for g_param_spec_types symbol linking error with MSVC

### PR DESCRIPTION
**Pull request by seungha-yang**
Regression from https://github.com/gtk-rs/glib/pull/688

cc: @sdroege

### CODE REVIEW

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by seungha-yang** [comment]
Sorry for this unclear commit message. (I don't fully understand why `#[no_mangle]` is needed here.

I've seen this build error with [gst-plugins-rs](https://gitlab.freedesktop.org/gstreamer/gst-plugins-rs)
```
libglib-20259afa4dec16ec.rlib(glib-20259afa4dec16ec.glib.8h6me5m7-cgu.8.rcgu.o) : error LNK2019: unresolved external symbol g_param_spec_types referenced in function _ZN92_$LT$glib..param_spec..param_spec_char..ParamSpecChar$u20$as$u20$glib..types..StaticType$GT$11static_type17h09c13f9e266f0171E
libglib-20259afa4dec16ec.rlib(glib-20259afa4dec16ec.glib.8h6me5m7-cgu.11.rcgu.o) : error LNK2001: unresolved external symbol g_param_spec_types
...
```

File: src/param_spec.rs [hunk 1]
```diff
@@ -612,6 +612,7 @@ pub trait ParamSpecType:
 }
 
 extern "C" {
+    #[no_mangle]
```
Comment (sdroege) [inline src/param_spec.rs:615]: It's unclear to me why this is actually required here but not for functions in `extern "C"` blocks and only on Windows/MSVC. The [nomicon](https://doc.rust-lang.org/nomicon/ffi.html#accessing-foreign-globals) also does not mention it.

A stackoverflow answer from @kornelski also mentions it though: https://stackoverflow.com/questions/45550387/how-can-i-access-a-c-global-variable-constant-in-rust-ffi

Comment (kornelski) [reply src/param_spec.rs:615]: I'm not sure if that code is necessary, or just happens to parse.

Comment (sdroege) [reply src/param_spec.rs:615]: Well here it fixes a linker error :) So some more digging is needed for what `rustc` is actually doing here.

**Comment by sdroege** [comment]
Curious:

```
warning: attribute should be applied to a function or static

   --> /home/travis/build/gtk-rs/glib/src/param_spec.rs:615:5

    |

615 |     #[no_mangle]

    |     ^^^^^^^^^^^^

616 |     pub static g_param_spec_types: *const glib_sys::GType;

    |     ------------------------------------------------------ not a function or static

    |

    = note: `#[warn(unused_attributes)]` on by default

    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
```

File: src/param_spec.rs (hunk 1 above)
Comment (seungha-yang) [reply src/param_spec.rs]: `#[no_mangle]` wasn't right fix. we need `#[link(name = "gobject-2.0")]` for the fix

**Review by sdroege — APPROVED** [review]

**Pull request merged**